tests/test_hubspot_bookmarks.py

```diff
@@ -48,24 +49,46 @@ def setUp(self):
         self.test_client = TestClient(self.get_properties()['start_date'])
 
     def create_test_data(self, expected_streams):
+        """
+        Creating more records(5) instead of 3 to get the update time to build the histogram - tdl-20939
+        Excluding workflows as it results in assertion failures with expected_pk and sync_pk at line#261
+        """
 
         self.expected_records = {stream: []
                                  for stream in expected_streams}
-
+        self.times=0
         for stream in expected_streams - {'contacts_by_company'}:
-            if stream == 'email_events':
-                email_records = self.test_client.create(stream, times=3)
+            self.minimum=sys.maxsize
+            self.maximum=-sys.maxsize-1
+            if stream == 'contacts': 
+                self.times=10
+            else:
+                self.times =3
+
+            if stream in 'email_events':
+                email_records = self.test_client.create(stream, self.times)
                 self.expected_records['email_events'] += email_records
             else:
                 # create records, one will be updated between syncs
-                for _ in range(3):
+                time_diff =0
+                for _ in range(self.times):
                     record = self.test_client.create(stream)
                     self.expected_records[stream] += record
+                    if stream in 'contacts':
+                        if self.test_client.time_difference < self.minimum:
+                            self.minimum = self.test_client.time_difference
+                        if self.test_client.time_difference > self.maximum:
+                            self.maximum = self.test_client.time_difference
+                        time_diff += self.test_client.time_difference 
+
+                if stream in 'contacts':
+                    self.avg = time_diff/5
+                    self.test_client.record_create_times[stream] = time_diff
```

Comment:
I don't think it is necessary to change the test to create more data.

It would be better to put all the data into your record_create_times in the client itself so that any test that creates this data by calling the client will have this data recorded.

Then once the test is finished you can use a tearDown method to analyze the data from the setup.  We could add this teardown to any test that uses the contacts stream to get data from multiple tests instead of just this one.

Reply:
Moved the code to record the times to client. And, added teardown method to print the result at the end.